### PR TITLE
582: Component zooming

### LINKED ISSUE #582: Allow zooming into a component
Something that was requested at the most recent ECP meeting.

### PR DESCRIPTION
**Pull request by DolicaAkelloEgwel**
### Issue

Closes #582 

### Description of work

Adds a feature for zooming into components. Improved a type hints in the instrument view. Sorted out the exception caused by deleting the sample.

### Acceptance Criteria 

Check that zooming works. Also check that the zoom button is disabled when a component isn't selected.

### UI tests

*How you have updated the UI tests document to fit your changes*

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: nexus_constructor/instrument_view.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def __init__(self, parent):
         super().__init__()
 
         self.root_entity = Qt3DCore.QEntity()
+        self.sample_entity = None
```
Comment (DominicOram) [inline nexus_constructor/instrument_view.py]: This isn't used

File: nexus_constructor/main_window.py [hunk 2]
```diff
@@ -153,6 +153,17 @@ def _set_up_tree_view(self):
...
+
+        self.zoom_action = QAction(
+            QIcon(os.path.join("ui", "zoom.svg")),
+            "Zoom To Component",
+            self.component_tree_view_tab,
+        )
+        self.zoom_action.triggered.connect(self.on_zoom_item)
+        self.zoom_action.setEnabled(False)
+        self.component_tool_bar.addSeparator()
+        self.component_tool_bar.addAction(self.zoom_action)
```
Comment (DominicOram) [inline nexus_constructor/main_window.py]: These changes are highlighting how much copy/pasted code there is in this function. Can we pull it out into an `add_action` method?

File: nexus_constructor/main_window.py [hunk 3]
```diff
@@ -208,14 +219,16 @@ def _load_current_entry():
...
+        if len(indices) != 1:
             self.delete_action.setEnabled(False)
             self.duplicate_action.setEnabled(False)
             self.new_rotation_action.setEnabled(False)
             self.new_translation_action.setEnabled(False)
             self.create_link_action.setEnabled(False)
+            self.zoom_action.setEnabled(False)
         else:
             selected_object = indices[0].internalPointer()
+            self.zoom_action.setEnabled(isinstance(selected_object, Component))
```
Comment (DominicOram) [inline nexus_constructor/main_window.py]: The logic for when buttons are visible or not isn't tested, can we add some unit tests here please?

Comment (DolicaAkelloEgwel) [reply nexus_constructor/main_window.py]: The main window looks untestable for the time being...

Comment (DolicaAkelloEgwel) [reply nexus_constructor/main_window.py]: It still says blocked in #404.

Comment (DominicOram) [reply nexus_constructor/main_window.py]: Ok, I think we need to do something about that, but it's outside the scope of this ticket. I've added a note to the linked ticket.

**Pull request merged**